atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/query/forming/AbstractQueryFormer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.async.query.forming;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.tritium.metrics.caffeine.CaffeineCacheStats;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+
+abstract class AbstractQueryFormer implements QueryFormer {
+
+    static final String CACHE_NAME_PREFIX = "query.async.prepared.statements.cache.metrics.";
+
+    static void registerCache(TaggedMetricRegistry taggedMetricRegistry, String name, Cache<String, String> cache) {
+        CaffeineCacheStats.registerCache(taggedMetricRegistry, cache, name);
+    }
+
+    static Cache<String, String> createCache(int cacheSize) {
+        return Caffeine.newBuilder().maximumSize(cacheSize).build();
+    }
+
+    // TODO (OStevan): prone to injection, fix this with some pattern match checking
+    static String normalizeName(String keyspace, TableReference tableReference) {
+        return keyspace + "." + tableReference.getQualifiedName();
+    }
```

Comment:
@felixdesouza can probably explain this in more detail, but I'm not sure I understand the need for inheritance here; this class has only static members. I'd extract a separate final utility class if you want the implementations of some of these methods to be shared.

I think if you need/want to refer to `QueryFormer` without knowing its concrete type, it would be preferable to just expose it at the top-level. It seems like e.g. `CachePerQueryFormer` can just implement `QueryFormer` directly (unless you have some other plans for how these classes are going to evolve).